### ISSUE TITLE
How To Make Operation Output Assessible

### ISSUE THREAD

**Issue by faraixyz** [Heydon/AMA#3]
Hey! I’m a fan of your stuff.

I’ve been working on a few front end projects. These typically take some input and provide some output (in the form of a table or even a link).

The one thing I’m not sure about is how to let screen reader users know that the operation has output.

These are a couple approaches I’ve tried.

* For this one app which tries to find an RSS feed url, I just change the focus to the resulting link
* for another app (a tax calculator) I used an atomic region on the table that shows the results.

I hope I’m not too confusing. Thanks in advance!

**Comment by Heydon** [comment]
@faraixyz Well, you've identified the only two decent approaches, so well done there!

I used to always advocate for a live region (or atomic region in your terminology 😉). However, some research with some screen reader users indicated they did not typically know what one was or, therefore, why they were hearing things without having _done_ anything.

This is one of the main issues with ARIA: it presents unfamiliar behaviors. Only widespread and longterm adoption can help this.

Anyway, one user actually thought focus had been moved when they heard the live region announcement, so they were expecting the behavior of your RSS app. So I would suggest doing this, probably.

If you do use a live region, be explicit with what's happening, and why they are hearing a message. Perhaps even unfocus and refocus the input to assure them their context has not changed (the role, value, etc would be reannounced after the live region announcement). 

Most of all, try testing with habitual screen reader users, preferably both blind and visual impaired. I hope that helps.
